src/runtime_src/doc/toc/xrt_native_apis.rst

```diff
@@ -58,195 +63,147 @@ Device and XCLBIN
+           auto output_buffer = xrt::bo(device, buffer_size_in_bytes, bank_grp_idx_1);
 
-           ....
-           ....
-           xrtBOFree(input_buffer);
-           xrtBOFree(output_buffer);
+In the above code ``xrt::bo`` buffer objects are created using the class's constructor. Please note the following 
 
-   
-The above code block shows 
+- As no special flags are used a regular buffer will be created.  
+- The second argument specifies the buffer size. 
+- The third argument should be used to specify the enumerated memory bank index (to specify the buffer location) where the buffer should be allocated. In this example, the ``xrt::kernel::group_id()`` member function is used to pass the memory bank index where the corresponding kernel arguments are (in the above example argument 0 and 1) connected. Instead of using the ``xrt::kernel::group_id()`` member function, the direct memory bank index (as observed in ``xbutil examine --report memory``) can be used as well. 
+  
+  
+Creating special Buffers
+************************
 
-    - Buffer allocation API ``xrtBOAlloc`` at lines 15,16
-    - Buffer deallocation API ``xrtBOFree`` at lines 23,24 
-    
-The various arguments of the API ``xrtBOAlloc`` are
+The ``xrt::bo()`` constructors accept multiple other buffer flags those are described using ``enum class`` argument with the following enumerator values
 
-    - Argument 1: The device on which the buffer should be allocated 
-    - Argument 2: The size (in bytes) of the buffer
-    - Argument 3: ``xrtBufferFlags``: Used to specify the buffer type, most commonly used types are
-       
-        - ``XRT_BO_FLAGS_NONE``: Regular Buffer
-        - ``XRT_BO_FLAGS_DEV_ONLY``: Device only Buffer (meant to be used only by the kernel). 
-        - ``XRT_BO_FLAGS_HOST_ONLY``: Host Only Buffer (buffers reside in the host memory directly transferred to/from the kernel)
-        - ``XRT_BO_FLAGS_P2P``: P2P Buffer, buffer for NVMe transfer
-        - ``XRT_BO_FLAGS_CACHEABLE``: Cacheable buffer can be used when host CPU frequently accessing the buffer (applicable for embedded platform). 
-        
-    - Argument 4:  ``xrtMemoryGroup``: Enumerated Memory Bank to specify the location on the device where the buffer should be allocated. The ``xrtMemoryGroup`` is obtained by the API ``xrtKernelArgGroupId`` as shown in line 15 (for more details of this API refer to the Kernel section).   
-    
+- ``xrt::bo::flags::normal``: Default, Regular Buffer
+- ``xrt::bo::flags::device_only``: Device only Buffer (meant to be used only by the kernel).
+- ``xrt::bo::flags::host_only``: Host Only Buffer (buffer resides in the host memory directly transferred to/from the kernel)
+- ``xrt::bo::flags::p2p``: P2P Buffer, buffer for NVMe transfer  
+- ``xrt::bo::flags::cacheable``: Cacheable buffer can be used when the host CPU frequently accessing the buffer (applicable for edge platform).
```

Comment:
Either lower case Buffer all over the place or be consistent with upper case buffer in Cacheable Buffer.  Personally, I don't really like Buffer as upper case.

Reply:
made all lowercase b buffer